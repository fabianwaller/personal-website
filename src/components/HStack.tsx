import { cn } from "@/lib/utils";

type HStackProps = {
  children: React.ReactNode;
  className?: string;
};

const HStack: React.FC<HStackProps> = ({ children, className }) => (
  <div
    className={cn([
      "flex flex-row flex-wrap items-center justify-start gap-x-4 gap-y-3",
      className,
    ])}
  >
    {children}
  </div>
);

export default HStack;
