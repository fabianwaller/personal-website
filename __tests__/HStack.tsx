import Header from "@/components/header";
import HStack from "@/components/HStack";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("HStack", () => {
  it("renders HStack unchanged", () => {
    const component = render(
      <HStack>
        <p>1</p>
        <p>2</p>
      </HStack>,
    );
    expect(component.asFragment()).toMatchSnapshot();
  });
});
