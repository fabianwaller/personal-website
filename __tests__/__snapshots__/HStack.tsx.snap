// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HStack renders HStack unchanged 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row flex-wrap items-center justify-start gap-x-4 gap-y-3"
  >
    <p>
      1
    </p>
    <p>
      2
    </p>
  </div>
</DocumentFragment>
`;
